Content.Server/Cloning/CloningSystem.cs

```diff
@@ -248,11 +252,22 @@ public bool TryCloning(EntityUid uid, EntityUid bodyToClone, Entity<MindComponen
             }
             // end of genetic damage checks
 
-            var mob = FetchAndSpawnMob(clonePod, pref, speciesPrototype, humanoid, bodyToClone, karmaBonus); //DeltaV Replaces CloneAppearance with Metem/Clone via FetchAndSpawnMob 
+            var mob = FetchAndSpawnMob(clonePod, pref, speciesPrototype, humanoid, bodyToClone, karmaBonus); //DeltaV Replaces CloneAppearance with Metem/Clone via FetchAndSpawnMob
 
             ///Nyano - Summary: adds the potential psionic trait to the reanimated mob.
             EnsureComp<PotentialPsionicComponent>(mob);
 
+            // Frontier - transfer of special components, e.g. small/big traits
```

Comment:
I need you to move "Transfer of Special Components" down into the FetchAndSpawnMob function, and then split it into two different conditions. We don't actually have Wizden cloning, we instead have a hybrid system that handles both Cloning and Metempsychosis based on checking for a bool. By placing this function up here, you are essentially making it so that when someone is reincarnated into a new body by the Metem machine, it transfers their tallness/shortness from old body to new body. Which is a big problem if for example their old body was a small vulpkanin and the new body is an Oni. 

This instead needs to be done only if the entity was cloned and not reincarnated. If they are instead reincarnated, an alternative behavior needs to be performed where it has a random chance of giving them the tallness/shortness(I would say at most 10% tall, 10% short, 80% normal).